contracts/strategies/qv-nft-tiered/QVNftTieredStrategy.sol

```diff
@@ -0,0 +1,176 @@
+            }
+        }
+    }
+
+    /// =========================
+    /// ==== View Functions =====
+    /// =========================
+
+    /// @notice Checks if the allocator is valid
+    /// @param _allocator The allocator address
+    /// @return true if the allocator is valid
+    function isValidAllocator(address _allocator) external view override returns (bool) {
+        return _isValidAllocator(_allocator);
+    }
+
+    /// =============================
+    /// ==== Internal Functions =====
+    /// =============================
+
+    function _isValidAllocator(address _allocator) internal view returns (bool) {
+        uint256 nftsLength = nfts.length;
+        for (uint256 i = 0; i < nftsLength;) {
+            if (nfts[i].balanceOf(_allocator) > 0) {
+                return true;
+            }
+
+            unchecked {
+                i++;
+            }
+        }
+
+        return false;
+    }
+
+    /// @notice Allocate votes to a recipient
+    /// @param _data The data
+    /// @param _sender The sender of the transaction
+    /// @dev Only the pool manager(s) can call this function
+    function _allocate(bytes memory _data, address _sender) internal override {
+        (address recipientId, ERC721 nft, uint256 nftId, uint256 voiceCreditsToAllocate) =
+            abi.decode(_data, (address, ERC721, uint256, uint256));
+
+        // check the voiceCreditsToAllocate is > 0
+        if (voiceCreditsToAllocate <= 0) {
+            revert INVALID();
+        }
+
+        // check the time periods for allocation
+        if (block.timestamp < allocationStartTime || block.timestamp > allocationEndTime) {
+            revert ALLOCATION_NOT_ACTIVE();
+        }
+
+        // check that the sender can allocate votes
+        if (!_isValidAllocator(_sender)) {
+            revert UNAUTHORIZED();
+        }
+
+        // spin up the structs in storage for updating
+        Recipient storage recipient = recipients[recipientId];
+        Allocator storage allocator = allocators[_sender];
+
+        if (voiceCreditsToAllocate + voiceCreditsUsedPerNftId[nft][nftId] > maxVoiceCreditsPerNft[nft]) {
+            revert INVALID();
+        }
+
+        uint256 creditsCastToRecipient = allocator.voiceCreditsCastToRecipient[recipientId];
+        uint256 votesCastToRecipient = allocator.votesCastToRecipient[recipientId];
+
+        uint256 totalCredits = voiceCreditsToAllocate + creditsCastToRecipient;
+        uint256 voteResult = _calculateVotes(totalCredits * 1e18);
+        voteResult -= votesCastToRecipient;
+        totalRecipientVotes += voteResult;
+        recipient.totalVotes += voteResult;
+
+        allocator.voiceCreditsCastToRecipient[recipientId] += totalCredits;
+        allocator.votesCastToRecipient[recipientId] += voteResult;
+
+        // update credits used by nftId
+        voiceCreditsUsedPerNftId[nft][nftId] += voiceCreditsToAllocate;
+
+        emit Allocated(_sender, voteResult, address(0), msg.sender);
```

Comment:
`emit Allocated(_sender, voteResult, address(0), msg.sender);`

should be
```solidity
// event Allocated(address indexed recipientId, uint256 amount, address token, address sender);
emit Allocated(recipientId, voteResult, address(0), _sender);
```

Reply:
This solves for when they use 1 NFT, what about an allocation using more than 1? Should the `nft` value passed in be an array?

Reply:
they can only use 1 nft per allocation
if you want to use multiple nfts, you need to call `batchAllocate` on `Allo.sol`